cloudsim-plus/src/main/java/org/cloudsimplus/faultinjection/HostFaultInjection.java

```diff
@@ -450,101 +451,135 @@ private void setVmToFailedAndCreateClone(Vm vm) {
         if (Host.NULL.equals(lastFailedHost)) {
             return;
         }
-        
-        final double recoveryTime = randomRecoveryTimeForVm.sample();
-        Log.printFormattedLine("\t# Time to recovery the fault: %.2f minutes", recoveryTime/60);    
 
-        vmRecoveryTimeMap.put(vm, recoveryTime);
-        
-        final DatacenterBroker broker = vm.getBroker();
-        final Vm clone = vmCloner.apply(vm);
-
-        List<Cloudlet> cloudlets = cloudletsCloner.apply(vm);
```

Comment:
This `List<Cloudlet> cloudlets = cloudletsCloner.apply(vm)` also must be moved as the line mentioned before, for the same reason. But in this case, this line must be after the `final Vm vmClone = cloneVm(broker, vm);`
Only after cloning the VM you have to clone the cloudlets.